packages/workspace/src/core/target-project-locator.ts

```diff
@@ -60,15 +60,73 @@ export class TargetProjectLocator {
       return this.findProjectOfResolvedModule(resolvedModule);
     }
 
-    if (this.paths && this.paths[normalizedImportExpr]) {
-      for (let p of this.paths[normalizedImportExpr]) {
+    const paths = this.findPaths(normalizedImportExpr);
+    if (paths) {
+      for (let p of paths) {
         const maybeResolvedProject = this.findProjectOfResolvedModule(p);
         if (maybeResolvedProject) {
           return maybeResolvedProject;
         }
       }
     }
 
+    // try to find npm package before using expensive typescript resolution
+    const npmProject = this.findNpmPackage(normalizedImportExpr);
+    if (npmProject) {
+      return npmProject;
+    }
+
+    // TODO(meeroslav): this block is probably obsolete
+    // and existed only because of the incomplete `paths` matching
+    // if import cannot be matched using tsconfig `paths` the compilation would fail anyway
+    const resolvedProject = this.resolveImportWithTypescript(
+      normalizedImportExpr,
+      filePath
+    );
+    if (resolvedProject) {
+      return resolvedProject;
+    }
+
+    // TODO(meeroslav): this block should be removed as it's going around typescript paths
+    // unless we want to support local JS projects
+    const importedProject = this.sortedWorkspaceProjects.find((p) => {
```

Comment:
This makes sense if we want to support JS projects or making sure for imports to be validated even if path is missing in `tsconfig`.